### DISCUSSION TITLE
Add SCIM to Outline

### DISCUSSION THREAD

**Discussion by routerino** [outline/outline#5821] [Ideas]
Was a comment [here](https://github.com/outline/outline/discussions/3779#discussioncomment-6962056), but putting in its own discussion for visibility.

[SCIM](https://simplecloud.info/) is a REST API schema for management of user accounts. This schema, when implemented, allows third parties to manage and sync user accounts on the behalf of outline. This is a feature built into many identity managers including keycloak, Azure AD, and [the long list on the SCIM site](https://simplecloud.info/). This allows for user management to happen in a vendor agnostic way.

There is a lot of overlap with outline's existing API functionality so hopefully it wouldn't be too onerous to extend the API to match SCIM capabilities.

**Comment by tommoor** [comment]
Definitely. 100% plan on implementing SCIM soon, but it won't be included in the community edition.

**Reply by ghost** [reply]
Has this made any progress? This was 18 months ago you said you were planning on implementing it "soon"

**Reply by FreeSynergy** [reply]
It is sad, that this feature isn't going to the community edition …
The bigger companies will pay for that, but the voluntary projects will have a disadvantage …

Make it, as you want, but this is not a nice step … (I can understand, why you do it - I always working for free and I know, how hard it is, to earn money - but if we do the same as all the others, how can we change the world??? 

For the wikis it would be a very good marketing thing, if you are the only one, with this feature … 
and the more it uses, the more you will get attention - look at Wiki.js … they are the first on the list, when you search for something like that …

Just think about it …

**Reply by gjermundksat** [reply]
@tommoor Any progress here? What is "soon"? :)

**Reply by tommoor** [reply]
Some progress, group sync is working with Google SSO in development - so the systems are all built and it needs extending to other providers.

**Comment by ghost** [comment]
For anyone interested, I wrote some PowerShell that will 'sort of' do provisioning and group membership. See: https://github.com/outline/outline/discussions/8614

From this you can write your own functions. We plan to take this and automate the listing of groups, then group memberships, then compare those against M365 group membership then add/remove as needed.
We'll just run it as a scheduled task. 

Not the most elegant, but I think it will get the job done until SCIM can actually be implemented for real.

**Comment by acul021** [comment]
Got tired of waiting for this and wrote a standalone adapter: https://github.com/acul021/scim-outline-adapter

It exposes SCIM 2.0 and maps it to Outline's REST API - users, groups, memberships, and team roles derived from group membership. We run it with authentik, should work with anything that speaks SCIM. Wrote up the details in https://github.com/outline/outline/discussions/12984
